src/serlo-editor/plugins/paste-hack/editor.tsx

```diff
@@ -101,7 +108,44 @@ export const PasteHackEditor: React.FunctionComponent<PasteHackPluginProps> = (
     }
   }
 
-  return <div>{renderDataImport()}</div>
+  return (
+    <>
+      {renderPluginToolbar()}
+      <div>{renderDataImport()}</div>
+    </>
+  )
+
+  function renderPluginToolbar() {
+    if (!focused) return null
+
+    const pasteHackButtonText = 'Import JSON data'
+
+    return (
+      <PluginToolbar
+        pluginType={EditorPluginType.PasteHack}
+        pluginControls={<PluginDefaultTools pluginId={id} />}
+        pluginSettings={
+          <>
+            <button
+              onClick={() => setShowSettingsModal(true)}
+              className="mr-2 rounded-md border border-gray-500 px-1 text-sm transition-all hover:bg-editor-primary-200 focus-visible:bg-editor-primary-200"
+            >
+              {pasteHackButtonText} <FaIcon icon={faPencilAlt} />
+            </button>
+            {showSettingsModal ? (
+              <ModalWithCloseButton
+                isOpen={showSettingsModal}
+                onCloseClick={() => setShowSettingsModal(false)}
+                className="!top-1/3 !max-w-xl"
+              >
+                {renderDataImport()}
+              </ModalWithCloseButton>
+            ) : null}
+          </>
+        }
```

Comment:
since this plugin replaces itself 😶‍🌫️, we don't need the settings again in the toolbar.